### DISCUSSION TITLE
What Lean code is generating the AST for the parser module?

### DISCUSSION THREAD

**Discussion by zaporter** [deepseek-ai/DeepSeek-Prover-V1.5#7] [Q&A] [answered]
Hi! Big fan of this repo and this project. This is a bit of a basic question, but I am stumped.

What lean code provides the AST? `lake exe repl` ([repo](https://github.com/leanprover-community/repl)) doesn't seem to support ast-retrieval. (even the quick example shows an empty ast object: https://github.com/deepseek-ai/DeepSeek-Prover-V1.5/blob/2c4ba9119eef74d0d611f494261b2c5bae98c69a/quick_start.py#L69)

Did the codebase originally use LeanDojo's `ExtractData.lean` to get the AST? (If it still does, where?? I can't find any references to LeanDojo or the required code to generate an AST (does mathlib4 have it builtin now?)).

Thanks for your time! (& for this cool paper / project!)

**Comment by xinhjBrant** [comment, marked as answer]
Thank you for your attention! The AST extraction code is originally based on LeanDojo's ExtractData.lean and references the following repository for the usage of this project: [ExtractData.lean](https://github.com/xinhjBrant/repl/blob/c6199a81de2a7e16cb27d6f85f56cff7043cd27f/REPL/ExtractData.lean#L461).
